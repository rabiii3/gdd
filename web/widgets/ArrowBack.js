import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { compose } from 'ramda';
import Icon from '@material-ui/core/Icon';
import IconButton from '@material-ui/core/IconButton';

export const ArrowBack = ({ history }) => (
  <IconButton onClick={() => history.goBack()}>
    <Icon className='fas fa-arrow-left' size="5x" />
  </IconButton>
);

ArrowBack.propTypes = {
  history: PropTypes.object.isRequired,
};
const enhance = compose(withRouter);
export default enhance(ArrowBack);
