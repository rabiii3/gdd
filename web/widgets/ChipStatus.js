import React from 'react';
import Chip from '@material-ui/core/Chip';
import PropTypes from 'prop-types';
import { statusLabel } from '../../lib/models/people';

const size2Dim = size => ({ LARGE: '230px', MEDIUM: '150px', SMALL: '60px' }[size]);

export const LARGE = 'LARGE';
export const MEDIUM = 'MEDIUM';
export const SMALL = 'SMALL';

const ChipStatus = ({ person, size = 'SMALL' }) => {
  const style = {
    width: size2Dim(size),
    height: 30,
  };
  const props = { style };

  return <Chip {...props} label={statusLabel(person)} color="primary" />;
};

ChipStatus.propTypes = {
  person: PropTypes.object,
  size: PropTypes.string,
};
export default ChipStatus;
