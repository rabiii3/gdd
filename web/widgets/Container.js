import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

const Styles = () => ({
  container: {
    height: '100vh',
  },
});

const Container = ({ classes, className, children }) => (
  <Grid container justify="center" className={classNames(className, classes.container)}>
    <Grid item xs={12}>
      {children}
    </Grid>
  </Grid>
);

Container.propTypes = {
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};
const enhance = compose(withStyles(Styles));
export default enhance(Container);
