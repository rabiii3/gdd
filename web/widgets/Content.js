import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

const Styles = {};

const Content = ({ classes, children }) => (
  <Grid container className={classes.content} justify="center" alignItems="flex-start">
    <Grid item xs={12}>
      {children}
    </Grid>
  </Grid>
);

Content.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node,
};
const enhance = compose(withStyles(Styles));
export default enhance(Content);
