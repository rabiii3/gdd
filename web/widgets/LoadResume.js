import React from 'react';
import PropTypes from 'prop-types';
import { withStateHandlers} from 'recompose';
import { Page, Document } from 'react-pdf';
import { compose } from 'ramda';
import { FormattedMessage } from 'react-intl';
import { SizeMe } from 'react-sizeme';
import Hidden from '@material-ui/core/Hidden/Hidden';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import Grid from '@material-ui/core/Grid/Grid';
import withStyles from '@material-ui/core/styles/withStyles';
import withMobileDialog from '@material-ui/core/withMobileDialog';
import messages from '../components/SaveResume/messages';

const styles = theme => ({
    progress: {
        margin: theme.spacing.unit * 2,
    },
    dialogSize: {
        [theme.breakpoints.up('sm')]: {
            minWidth: '42rem',
            height: '60rem',
            overflowX: 'hidden',
        },
        overflowX: 'auto',
    },
});

const LoadResume = ({ classes, file, open, close, numPages, onDocumentLoaded, download, fullScreen }) => (
    <Dialog
        open={open}
        onClose={close}
        fullScreen={fullScreen}
    >
        <DialogContent className={classes.dialogSize}>
            <Document
                loading={
                    <Grid container justify="center" alignItems="center">
                        <Grid item>
                            <CircularProgress className={classes.progress} />
                        </Grid>
                    </Grid>
                }
                onLoadSuccess={onDocumentLoaded}
                file={file}
            >
                <Hidden xsDown>
                    {Array.from(
                        new Array(numPages),
                        (el, index) => (<Page
                            key={`page_${index + 1}`}
                            pageNumber={index + 1}
                        />
                        ))}
                </Hidden>
                <Hidden smUp>
                    {Array.from(
                        new Array(numPages),
                        (el, index) => (
                            <SizeMe key={`page_${index + 1}`}>
                                {() => <Page
                                    pageNumber={index + 1}
                                    width={350}
                                />}
                            </SizeMe>
                        ))}
                </Hidden>
            </Document>
        </DialogContent>
        <DialogActions>
            <Button onClick={download} color="primary" disabled={!numPages}>
                <FormattedMessage {...messages.downloadFile} />
            </Button>
            <Button onClick={close} color="primary">
                <FormattedMessage {...messages.close} />
            </Button>
        </DialogActions>
    </Dialog >
);

LoadResume.propTypes = {
    classes: PropTypes.object,
    open: PropTypes.bool,
    close: PropTypes.func,
    getDocumentLoad: PropTypes.func,
    numPages: PropTypes.number,
    file: PropTypes.object,
    download: PropTypes.func,
    fullScreen: PropTypes.bool.isRequired,
    onDocumentLoaded: PropTypes.func,
};

const withPageHandlers = withStateHandlers(
  { numPages: 0 },
  {
    onDocumentLoaded: () => ({ numPages }) => ({ numPages }),
  }
);


export const enhance = compose(
  withPageHandlers,
  withStyles(styles), 
  withMobileDialog()
);

export default enhance(LoadResume);
