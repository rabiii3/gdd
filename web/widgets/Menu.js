import React from 'react';
import PropTypes from 'prop-types';
import Icon from '@material-ui/core/Icon';
import Popover from '@material-ui/core/Popover';
import { compose } from 'ramda';
import { withState, withHandlers } from 'recompose';
import IconButton from '@material-ui/core/IconButton';

const size2Dim = size => ({ LARGE: '32px', MEDIUM: '24px', SMALL: '18px' }[size]);
export const LARGE = 'LARGE';
export const MEDIUM = 'MEDIUM';
export const SMALL = 'SMALL';

export const Menu = ({ openUserMenu, closeUserMenu, userMenuAnchor, children, size = 'SMALL' }) => {
  const style = {
    fontSize: size2Dim(size)
  };
  const props = { style };
  return (
    <div>
      <IconButton onClick={openUserMenu}>
        <Icon className="fas fa-ellipsis-v" {...props} />
      </IconButton>
      <Popover
        id="menu-appbar"
        anchorEl={userMenuAnchor}
        open={Boolean(userMenuAnchor)}
        onClose={closeUserMenu}
        getContentAnchorEl={null}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'right',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
      >
        {children}
      </Popover>
    </div>
  );
};
Menu.propTypes = {
  openUserMenu: PropTypes.func.isRequired,
  closeUserMenu: PropTypes.func.isRequired,
  userMenuAnchor: PropTypes.object,
  children: PropTypes.node.isRequired,
  size: PropTypes.string,
};

const withMenu = withState('userMenuAnchor', 'toggleMenu', null);
const withMenuHandlers = withHandlers({
  openUserMenu: ({ toggleMenu }) => event => toggleMenu(event.currentTarget),
  closeUserMenu: ({ toggleMenu }) => () => toggleMenu(null),
});
const enhance = compose(
  withMenu,
  withMenuHandlers,
);
export default enhance(Menu);
