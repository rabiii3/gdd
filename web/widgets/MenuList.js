import React from 'react';
import { compose } from 'recompose';
import { withStyles } from '@material-ui/core';
import PropTypes from 'prop-types';
import MenuList from '@material-ui/core/MenuList';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import DeleteIcon from '@material-ui/icons/Delete';
import { FormattedMessage } from 'react-intl';
import messages from '../components/App/messages';

const Styles = theme => ({
  menuItem: {
    '&:focus': {
      backgroundColor: theme.palette.primary.main,
      '& $primary, & $icon': {
        color: theme.palette.common.white,
      },
    },
  },
  primary: {},
});

const MenuUserList = ({ classes, actions, person }) => (
  <Paper>
    <MenuList>
      <MenuItem className={classes.menuItem}>
        <ListItemIcon onClick={() => actions.deletePerson(person)}  >
          <DeleteIcon />
          <ListItemText
            primary={<FormattedMessage {...messages.delete} />}
          />
        </ListItemIcon>

      </MenuItem>
    </MenuList>
  </Paper>
);

MenuUserList.propTypes = {
  classes: PropTypes.object.isRequired,
  actions: PropTypes.object,
  person: PropTypes.object,
};

const enhance = compose(withStyles(Styles));
export default enhance(MenuUserList);
