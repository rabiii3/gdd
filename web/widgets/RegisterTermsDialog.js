import React from 'react';
import { Typography } from '@material-ui/core';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { compose } from 'ramda';
import { FormattedMessage } from 'react-intl';
import { withHandlers, withState } from 'recompose';
import messages from './messages';

const Styles = {
  terms: {
    color: 'blue',
    textDecoration: 'underline',
    cursor: 'pointer',
  },
};
const RegisterTermsDialog = ({ classes, closeTermsDialog, openTermsDialog, isDialogOpened }) => (
  <Grid>
    <Typography className={classes.terms} onClick={openTermsDialog}>
      <FormattedMessage {...messages.terms} />
    </Typography>
    <Dialog open={isDialogOpened} onClose={closeTermsDialog} scroll="paper" aria-labelledby="scroll-dialog-title">
      <DialogTitle id="scroll-dialog-title">
        {' '}
        <FormattedMessage {...messages.terms} />
      </DialogTitle>
      <DialogContent>
        <DialogContentText>
          <FormattedMessage {...messages.acceptTermsContent} />
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={closeTermsDialog} color="primary">
          Close
        </Button>
      </DialogActions>
    </Dialog>
  </Grid>
);
RegisterTermsDialog.propTypes = {
  openTermsDialog: PropTypes.func,
  closeTermsDialog: PropTypes.func,
  isDialogOpened: PropTypes.bool,
  classes: PropTypes.object,
};

const withDialog = withState('isDialogOpened', 'toggleDialog', false);
const withDialogHandlers = withHandlers({
  openTermsDialog: ({ toggleDialog }) => () => toggleDialog(true),
  closeTermsDialog: ({ toggleDialog }) => () => toggleDialog(false),
});
const enhance = compose(
  withStyles(Styles),
  withDialog,
  withDialogHandlers,
);
export default enhance(RegisterTermsDialog);
