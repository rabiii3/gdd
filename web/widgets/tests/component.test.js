import React from 'react';
import { shallow } from 'enzyme'; // eslint-disable-line
import Container from '../Container';
import Content from '../Content';
import ArrowBack from '../ArrowBack';
import GridWidget from '../Grid';
import Menu from '../Menu';
import MenuList from '../MenuList';
import UserName from '../Name';
import ResumeIcon from '../ResumeIcon';
import Spacer from '../Spacer';
import ChipStatus from '../ChipStatus';
import ResumeStatus from '../ResumeStatus';
import RegisterTermsDialog from '../RegisterTermsDialog';
import RoleChip from '../RoleChip';
import { SkillsChip, FilterMenu } from '..';
import LoadResume from '../LoadResume';

describe('<Container />', () => {
  const props = {
    children: <div />,
    classes: {},
  };
  test('should render a <Container />', () => {
    const tree = shallow(<Container {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
describe('<Content />', () => {
  const props = {
    children: <div />,
    classes: {},
  };
  test('should render a <Content />', () => {
    const tree = shallow(<Content {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<ArrowBack />', () => {
  const props = {
    history: {},
  };
  test('should render an <ArrowBack />', () => {
    const tree = shallow(<ArrowBack {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<Grid />', () => {
  const props = {
    children: <div />,
  };
  test('should render a <Grid />', () => {
    const tree = shallow(<GridWidget {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<Menu ><MenuList/></Menu>', () => {
  const props = {
    Menu: {
      children: <div />,
      openUserMenu: jest.fn(),
      closeUserMenu: jest.fn(),
      userMenuAnchor: {},
    },
    MenuList: {
      classes: {},
    },
  };
  test('should render a <Menu /> and <MenuList/>', () => {
    const tree = shallow(
      <Menu {...props.Menu}>
        <MenuList {...props.MenuList} />
      </Menu>,
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('<UserName />', () => {
  const props = {
    user: {},
    classes: {},
  };
  test('should render a <UserName />', () => {
    const tree = shallow(<UserName {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<ResumeIcon />', () => {
  test('should render a <ResumeIcon />', () => {
    const wrapper = shallow(<ResumeIcon />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Spacer />', () => {
  const props = {
    size: 'SMALL',
  };
  test('should render a <Spacer />', () => {
    const tree = shallow(<Spacer {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<ChipStatus />', () => {
  const props = {
    size: 'SMALL',
    person: {},
  };
  test('should render a <ChipStatus />', () => {
    const tree = shallow(<ChipStatus {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<ResumeStatus />', () => {
  const props = {
    size: 'SMALL',
    resume: {},
    action: jest.fn(),
    location: {},
  };
  test('should render a <ResumeStatus />', () => {
    const tree = shallow(<ResumeStatus {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<RegisterTermsDialog />', () => {
  const props = {
    openTermsDialog: jest.fn(),
    closeTermsDialog: jest.fn(),
    isDialogOpened: false,
    classes: {},
  };
  test('should render a <RegisterTermsDialog />', () => {
    const tree = shallow(<RegisterTermsDialog {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<RoleChip />', () => {
  const props = {
    data: [],
    size: 'SMALL',
  };
  test('should render a <RoleChip />', () => {
    const tree = shallow(<RoleChip {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<SkillsChip />', () => {
  const props = {
    data: [],
    size: 'SMALL',
    action: jest.fn(),
    location: {},
  };
  test('should render a <SkillsChip />', () => {
    const tree = shallow(<SkillsChip {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<LoadResume />', () => {
  const props = {
    classes: {},
    open: true,
    close: jest.fn(),
    getDocumentLoad: jest.fn(),
    numPages: {},
    token: "",
    resume: {},
    download: jest.fn(),
    fullScreen: true,
  };
  test('should render a <LoadResume />', () => {
    const tree = shallow(<LoadResume {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('<FilterMenu />', () => {
  const props = {
    classes: {},
    closeMenu: jest.fn(),
    isOpen: {},
    data: [],
    action: jest.fn(),
    statusFilter: '',
    allResumes: 10,
  };
  test('should render a <FilterMenu />', () => {
    const tree = shallow(<FilterMenu {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
