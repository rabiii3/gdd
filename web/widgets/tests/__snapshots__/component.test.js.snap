// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowBack /> should render an <ArrowBack /> 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`<ChipStatus /> should render a <ChipStatus /> 1`] = `
<WithStyles(Chip)
  color="primary"
  style={
    Object {
      "height": 30,
      "width": "60px",
    }
  }
/>
`;

exports[`<Container /> should render a <Container /> 1`] = `
<Container
  classes={
    Object {
      "container": "Container-container-1",
    }
  }
>
  <div />
</Container>
`;

exports[`<Content /> should render a <Content /> 1`] = `
<Content
  classes={Object {}}
>
  <div />
</Content>
`;

exports[`<FilterMenu /> should render a <FilterMenu /> 1`] = `
<FilterMenu
  action={[MockFunction]}
  allResumes={10}
  classes={
    Object {
      "badge": "FilterMenu-badge-7",
      "menuItem": "FilterMenu-menuItem-8",
    }
  }
  closeMenu={[MockFunction]}
  data={Array []}
  isOpen={Object {}}
  statusFilter=""
/>
`;

exports[`<Grid /> should render a <Grid /> 1`] = `
<WithStyles(Grid)
  item={true}
  xs={12}
>
  <WithStyles(Grid)
    container={true}
    wrap="nowrap"
  >
    <div />
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;

exports[`<LoadResume /> should render a <LoadResume /> 1`] = `
<WithStyles(WithTheme(WithWidth(WithMobileDialog)))
  classes={Object {}}
  close={[MockFunction]}
  download={[MockFunction]}
  fullScreen={true}
  getDocumentLoad={[MockFunction]}
  numPages={0}
  onDocumentLoaded={[Function]}
  open={true}
  resume={Object {}}
  token=""
/>
`;

exports[`<Menu ><MenuList/></Menu> should render a <Menu /> and <MenuList/> 1`] = `
<withHandlers(Menu)
  closeUserMenu={[MockFunction]}
  openUserMenu={[MockFunction]}
  toggleMenu={[Function]}
  userMenuAnchor={null}
>
  <WithStyles(MenuUserList)
    classes={Object {}}
  />
</withHandlers(Menu)>
`;

exports[`<RegisterTermsDialog /> should render a <RegisterTermsDialog /> 1`] = `
<withState(withHandlers(RegisterTermsDialog))
  classes={
    Object {
      "terms": "withState-withHandlers-RegisterTermsDialog---terms-4",
    }
  }
  closeTermsDialog={[MockFunction]}
  isDialogOpened={false}
  openTermsDialog={[MockFunction]}
/>
`;

exports[`<ResumeIcon /> should render a <ResumeIcon /> 1`] = `
<ResumeIcon
  classes={
    Object {
      "root": "ResumeIcon-root-3",
    }
  }
/>
`;

exports[`<ResumeStatus /> should render a <ResumeStatus /> 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<RoleChip /> should render a <RoleChip /> 1`] = `null`;

exports[`<SkillsChip /> should render a <SkillsChip /> 1`] = `
<EnhancedComponent
  action={[MockFunction]}
  classes={
    Object {
      "chip": "EnhancedComponent-chip-6",
      "root": "EnhancedComponent-root-5",
    }
  }
  data={Array []}
  location={Object {}}
  size="SMALL"
/>
`;

exports[`<Spacer /> should render a <Spacer /> 1`] = `
<div
  style={
    Object {
      "marginLeft": "3px",
      "marginRight": "3px",
    }
  }
/>
`;

exports[`<UserName /> should render a <UserName /> 1`] = `
<UserName
  classes={
    Object {
      "name": "UserName-name-2",
    }
  }
  user={Object {}}
/>
`;
